crates/net/network/src/config.rs

```diff
@@ -260,8 +260,8 @@ impl<C> NetworkConfigBuilder<C> {
     }
 
     /// Sets the discovery on or off.
-    pub fn set_discovery(mut self, disable_discovery: bool) -> Self {
-        self.discovery_v4_builder.set_discovery(disable_discovery);
+    pub fn disable_discovery(mut self) -> Self {
```

Comment:
let's keep this but also add `set_discovery(bool)` this makes the builder more convenient, see node/mod.rs